devops/azure-pipelines/kubernetes-template.yml

```diff
@@ -1,16 +1,28 @@
 parameters:
   - name: 'jobName'
     type: string
-  - name: 'serviceConnection' #Kubernetes Service Connection name
+  - name: 'branchMain' #Main branch name
+    type: string
+  - name: 'serviceConnectionMain' #Kubernetes Service Connection name
+    type: string
+  - name: 'serviceConnectionDev' #Kubernetes Service Connection name
     type: string
   - name: 'manifestPath' #Kubernetes manifest file path
     type: string
 jobs:
   - job: ${{ parameters.jobName }}
     steps:
       - task: KubernetesManifest@0
+        condition: eq(variables['Build.SourceBranch'], '${{ parameters.branchMain }}')
+        inputs:
+          action: 'deploy'
+          kubernetesServiceConnection: '${{ parameters.serviceConnectionMain }}'
+          manifests: '${{ parameters.manifestPath }}'
+          rolloutStatusTimeout: '10'
+      - task: KubernetesManifest@0
+        condition: ne(variables['Build.SourceBranch'], '${{ parameters.branchMain }}')
         inputs:
           action: 'deploy'
-          kubernetesServiceConnection: '${{ parameters.serviceConnection }}'
+          kubernetesServiceConnection: '${{ parameters.serviceConnectionDev }}'
```

Comment:
- every step should have a displayName
- consider having this "dev or main" switch be handled at the pipeline level, leaving this file free to just think about one step with no conditions. you could use template syntax, like [this](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#conditionally-set-a-task-input), to reduce duplication in either case. it should work at the pipeline level too.

Reply:
displayName stuff added. working on conditional inputs on different branch, WIP not yet pushed

Reply:
displayNames added, change in conditional logic works.